tests/unit/test_tiler.py

```diff
@@ -340,12 +341,27 @@ def it_can_generate_random_tiles_even_when_coords_are_not_valid(
         assert generated_tiles[0][1] == CP(0, 10, 0, 10)
         assert isinstance(generated_tiles[0][0], Tile)
 
-    def it_can_extract_random_tiles(self, request, tmpdir):
+    @pytest.mark.parametrize(
+        "path_type_transform",
+        [
+            (str),
+            (Path),
+        ],
+    )
+    def it_can_extract_random_tiles(
+        self,
+        request,
+        tmpdir,
+        path_type_transform,
+    ):
```

Comment:
I don't see the need to have both cases here. Considering that this change is a `Slide` change and in here we're testing the tiler so this parametrisation is pointless IMHO. You already have a test with both str and Path in the slides unit tests ... so I'd revert the change here

Reply:
> You already have a test with both str and Path in the slides unit tests

yes but since we are lazy loading the slide, that test is not meaningful if the goal is checking if histolab can handle `pathlib.Path` before opening a slide with openslide

I'm adding another unit test that opens a slide from disk to check the `._wsi` attribute and I'm reverting this change